docs/conf.py

```diff
@@ -8,12 +8,36 @@
 from sys import path
 
 from setuptools_scm import get_version
+from setuptools_scm.git import fetch_on_shallow
+from setuptools_scm.git import parse
 
 
+# -- Special accommodations for RTD ------------------------------------------
+
+
+def parse_with_fetch(*args, **kwargs) -> str:
+    """If the repo is found to be shallow, fetch a full.
+
+    By default, RTD does a fetch --limit 50, if a tag is not
+    present in the last 50 commits, the version reported by setuptools_scm
+    will be incorrect and appears as ``v0.1.dev...`` in the towncrier changelog.
+    Another approach is to enable ``DONT_SHALLOW_CLONE`` for the repo
+    https://docs.readthedocs.io/en/stable/feature-flags.html#feature-flags
+    This was done for ansible-navigator on the day of this commit.
+
+    :param args: The arguments
+    :param kwargs: The keyword arguments
+    :returns: The parsed version
+    """
```

Comment:
Since it's not a lambda, we can properly declare the expectations here now:
```suggestion
    """
    assert 'pre_parse' not in kwargs
```

Reply:
interesting..........I guess good insurance

Reply:
Not just insurance but also expectation declaration per contract programming principles :)